test/extended/image_ecosystem/mongodb_replica_petset.go

```diff
@@ -0,0 +1,112 @@
+package image_ecosystem
+
+import (
+	"fmt"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	dbutil "github.com/openshift/origin/test/extended/util/db"
+)
+
+var _ = g.Describe("[image_ecosystem][mongodb][Slow] openshift mongodb replication (with petset)", func() {
+	defer g.GinkgoRecover()
+
+	const templatePath = "https://raw.githubusercontent.com/sclorg/mongodb-container/master/examples/petset/mongodb-petset-persistent.yaml"
+
+	oc := exutil.NewCLI("mongodb-petset-replica", exutil.KubeConfigPath()).Verbose()
+
+	g.Describe("creating from a template", func() {
+		g.It(fmt.Sprintf("should process and create the %q template", templatePath), func() {
+			oc.SetOutputDir(exutil.TestContext.OutputDir)
+
+			g.By("creating persistent volumes")
+			_, err := exutil.SetupHostPathVolumes(
+				oc.AdminKubeREST().PersistentVolumes(),
+				oc.Namespace(),
+				"256Mi",
+				3,
+			)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			defer func() {
+				// We're removing only PVs because all other things will be removed
+				// together with namespace.
+				// TODO: unfortunately it doesn't remove the directories on the filesystem.
+				// Need to figure out how to fix it.
+				err := exutil.CleanupHostPathVolumes(oc.AdminKubeREST().PersistentVolumes(), oc.Namespace())
```

Comment:
can't cleanupPersistentVolume get the hostPath from the PV and rm -rf it?  it's already iterating the PV objects.